pkg/storegateway/bucket.go

```diff
@@ -719,6 +729,21 @@ func blockSeries(
 		return nil, nil, errors.Wrap(err, "load chunks")
 	}
 
+	if ignoreNativeHistograms {
+		for i := len(res) - 1; i >= 0; i-- {
+			for j := len(res[i].chks) - 1; j >= 0; j-- {
+				if res[i].chks[j].Raw == nil {
+					res[i].refs = slices.Delete(res[i].refs, j, j+1)
+					res[i].chks = slices.Delete(res[i].chks, j, j+1)
```

Comment:
On average this will end up copying half of the elements in the slice for each native histogram chunk. I think we can iterate once and shift out the chunks with native histograms. This should copy at most every element once.

This is another piece of code which does a similar in-place shift for items in `g.keys` that don't match certain criterion  https://github.com/grafana/mimir/blob/3fde52076afa3460423591f461ea154f412fa068/pkg/storegateway/bucket_index_postings.go#L102-L115

And you can do a similar thing for series that are without chunks.

Reply:
Yep, I was definitely concerned about the performance here. Thanks for the suggestion--that sounds like it'll do the trick!